internal/cache/util/util_test.go

```diff
@@ -233,3 +234,37 @@ func (ut *utilTest) Test_IsCacheHandleValid_False() {
 		ExpectFalse(IsCacheHandleInvalid(errors.New(errMsg)))
 	}
 }
+
+func TestCheckPermissionsForShouldNotReturnAnyErrorWhenDirectoryExists(t *testing.T) {
+	base := "./" + string(testutil.GenerateRandomBytes(4))
+	dirPath := base + "/" + "path/cachedir"
+	dirCreationErr := os.MkdirAll(dirPath, 0700)
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
+	ExpectEq(nil, err)
+}
+
+func TestCheckPermissionsForShouldNotReturnAnyErrorWhenDirectoryCanBeCreated(t *testing.T) {
+	base := "./" + string(testutil.GenerateRandomBytes(4))
+	dirPath := base + "/" + "path/cachedir"
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, err)
+}
+
+func TestCheckPermissionsForShouldReturnErrorWhenDirectoryDoesNotHavePermissions(t *testing.T) {
+	dirPath := "./" + string(testutil.GenerateRandomBytes(4))
+	dirCreationErr := os.MkdirAll(dirPath, 0444)
+	defer os.RemoveAll(dirPath)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
+	ExpectNe(nil, err)
+	ExpectTrue(strings.Contains(err.Error(), "error creating file at directory ("+dirPath+"), error"))
```

Comment:
what's the "error" in the end of string ? is this returned by f, err := fsutil.AnonymousFile(dirPath) ?

Reply:
yes its returned when permission issue is there, I just wanted to assert that annonymous error string is returned not the create dir error hence added just contains, expected is not complete, since starting string confirms that error is from annoymous file